### ISSUE TITLE
Randomised eigensolver loses negative eigenvalues and mishandles zero and complex systems

### ISSUE THREAD

**Issue by sylvesterkaczmarek** [gchq/coreax#1236]
Reproduced on `main` at `f954211` using the locked Python 3.12 dependencies.

```python
import jax.numpy as jnp
import jax.random as jr
from coreax.least_squares import RandomisedEigendecompositionSolver

solver = RandomisedEigendecompositionSolver(jr.key(1), oversampling_parameter=2)
identity = jnp.eye(2)
for matrix in [jnp.zeros((2, 2)), jnp.diag(jnp.array([-4., 2.])),
               jnp.array([[2., 1j], [-1j, 3.]])]:
    print(solver.solve(matrix, 0., identity, identity))
    print(jnp.linalg.pinv(matrix, hermitian=True))
```

The zero system returns NaNs instead of zeros. The indefinite diagonal system loses the `-0.25` inverse entry. The complex Hermitian system gives a non-Hermitian, incorrect inverse even though the sketch spans the entire matrix.

The inversion mask compares signed eigenvalues with the largest signed eigenvalue and includes exact zeros when the cutoff is zero. Range iteration, projection and reconstruction also use ordinary transposes where the documented Hermitian input requires conjugate transposes.

Expected: truncate by eigenvalue magnitude, preserve retained signs, exclude exact zero eigenvalues, and retain complex Hermitian structure. Full-rank sketches should agree with the Moore-Penrose inverse, including under JIT and batched solves.

I have deterministic reproductions and failing regression tests, and am preparing a focused fix.
